lock-impl/src/main/java/com/palantir/lock/impl/LegacyTimelockService.java

```diff
@@ -103,23 +106,28 @@ public long getImmutableTimestamp() {
     }
 
     @Override
-    public LockRefreshTokenV2Adapter lock(LockRequestV2 request) {
+    public Optional<LockTokenV2> lock(LockRequestV2 request) {
         LockRequest legacyRequest = toLegacyLockRequest(request);
 
         LockRefreshToken legacyToken = lockAnonymous(legacyRequest);
-        return new LockRefreshTokenV2Adapter(legacyToken);
+        return Optional.ofNullable(legacyToken).map(LockRefreshTokenV2Adapter::new);
     }
 
     @Override
-    public void waitForLocks(WaitForLocksRequest request) {
+    public boolean waitForLocks(WaitForLocksRequest request) {
         LockRequest legacyRequest = toLegacyWaitForLocksRequest(request.getLockDescriptors());
 
         lockAnonymous(legacyRequest);
+        return true;
     }
 
     private LockRequest toLegacyLockRequest(LockRequestV2 request) {
        SortedMap<LockDescriptor, LockMode> locks = buildLockMap(request.getLockDescriptors(), LockMode.WRITE);
-       return LockRequest.builder(locks).build();
+       LockRequest.Builder builder =  LockRequest.builder(locks);
+       if (request.getAcquireTimeoutMs() < Long.MAX_VALUE) {
```

Comment:
alternatively, we can just remove the ability to block indefinitely, and just use whatever timeout is provided.

only difference between block indefinitely and block for a really long time is that `RemoteLockService` interrupts indefinite blockers if leadership is lost